cpp/src/arrow/dataset/scanner.cc

```diff
@@ -244,6 +244,8 @@ Status ScannerBuilder::UseThreads(bool use_threads) {
   return Status::OK();
 }
 
+void ScannerBuilder::UseAsync(bool use_async) { scan_options_->use_async = use_async; }
```

Comment:
nit: all the other methods return Status (even if it's a bit pointless)

Reply:
Fair enough.  I changed it to return a Status.